### ISSUE TITLE
BigQuery: Option to have L054 ignore GROUP BY/ORDER BY clauses in aggegate functions (e.g. ARRAY_AGG())

### ISSUE THREAD

**Issue by jdub** [sqlfluff/sqlfluff#3649]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

In this query, an L054 (Inconsistent column references in 'GROUP BY/ORDER BY' clauses.) is detected at the `order by` within the `array_agg` array function on line 4. The [L054 documentation](https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.core.rules.Rule_L054) says "ORDER BY clauses from WINDOW clauses are ignored by this rule", so it probably doesn't take this odd `order by` inside an array function into account.

```sql
    select
        user_uuid,
        max(started_at) as updated_at,
        to_json_string(array_agg(product_id order by started_at desc)) as products

    from purchased
    group by 1
```

### Expected Behaviour

L054 should probably ignore order by in array functions.

### Observed Behaviour

See "What Happened".

### How to reproduce

stick that query in a file :-)

### Dialect

bigquery

### Version

1.2.1

### Configuration

```ini
[sqlfluff]
dialect = bigquery
templater = dbt

# https://docs.sqlfluff.com/en/stable/rules.html
# L016: Line is too long
# L034: Use wildcards then simple targets before calculations and aggregates in select statements
exclude_rules = L016,L034

[sqlfluff:rules:L014]
extended_capitalisation_policy = lower
ignore_words = _TABLE_SUFFIX

[sqlfluff:rules:L064]
preferred_quoted_literal_style = single_quotes
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
> In this query, an L054 (Inconsistent column references in 'GROUP BY/ORDER BY' clauses.) is detected at the `order by` within the `array_agg` window function on line 4, even though the [L054 documentation](https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.core.rules.Rule_L054) says "ORDER BY clauses from WINDOW clauses are ignored by this rule".
> 
> Perhaps `array_agg` is not understood to be a window function in the BigQuery dialect?

I'm not sure if [array functions](https://cloud.google.com/bigquery/docs/reference/standard-sql/aggregate_functions#array_agg) called like that technically are [bigquery window functions](https://cloud.google.com/bigquery/docs/reference/standard-sql/window-function-calls)?

Compare these two bits of SQL:

```sql
to_json_string(array_agg(product_id order by started_at desc)) as products
to_json_string(array_agg(product_id) OVER (order by started_at desc)) as products
```

The first (yours) isn't a window function by [bigquery's definition](https://cloud.google.com/bigquery/docs/reference/standard-sql/window-function-calls) as it requires an `OVER` clause and flags under L054, but the second is a windowing function call and I've verified does not flag L054.

So that's why it doesn't work. Whether L054 should be enhanced to also ignore aggregation functions as well is a good question.

**Comment by jdub** [comment]
Oh, yeah! Well, that explains why it wasn't taken into account.

**Comment by jdub** [comment]
I've updated the title and body of the report.

**Comment by nishantnathhoney** [comment]
👍 +1 for consideration of adding an additional flag to skip enforcing this rule in window functions. Its common practice to be explicit in window functions which are often short & hence makes it readable but regular group by/order by clauses can be lengthy where positional reference/implicit reference is preferred.

**Comment by tunetheweb** [comment]
L054 already skips enforcing this role in windowing functions as noted in the docs: https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.core.rules.Rule_L054
This is done for precisely the reasons you note.

The issue here is an array function is technically not a windowing function, so that skipping logic needs to be increased to include that.

**Comment by nishantnathhoney** [comment]
> L054 already skips enforcing this role in windowing functions as noted in the docs: https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.core.rules.Rule_L054 This is done for precisely the reasons you note.
> 
> The issue here is an array function is technically not a windowing function, so that skipping logic needs to be increased to include that.

@tunetheweb apologies, you are right, I meant [aggregate functions](https://cloud.google.com/bigquery/docs/reference/standard-sql/aggregate_functions)

**Comment by remilepriol** [comment]
Note that this is not specific to BigQuery. Many engines support array_agg (presto, postgre, snowflake ...)
